src/ArmTemplates/Extractor/EntityExtractors/ParametersExtractor.cs

```diff
@@ -119,6 +120,39 @@ async Task AddServiceUrlParameterAsync()
                 parameters.Add(ParameterNames.ServiceUrl, new TemplateObjectParameterProperties() { Value = serviceUrls });
             }
 
+            async Task AddApiOauth2ScopeParameterAsync()
+            {
+                if (!extractorParameters.ParametrizeApiOauth2Scope)
+                {
+                    return;
+                }
+
+                var apiOauth2Scopes = new Dictionary<string, string>();
+                foreach (var apiName in apisToExtract)
+                {
+                    var apiDetails = await this.apisClient.GetSingleAsync(apiName, extractorParameters);
+
+                    if (apiDetails.Properties.AuthenticationSettings?.OAuth2 is not null)
+                    {
+                        string apiOAuthScope;
```

Comment:
you can use [ternary](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator) here, because it seems like a simple evaluation:
```suggestion
                        var apiOAuthScope = extractorParameters.ApiOauth2ScopeParameters is null
                            ? apiDetails.Properties.AuthenticationSettings.OAuth2?.Scope
                            : extractorParameters.ApiOauth2ScopeParameters.FirstOrDefault(x => x.ApiName.Equals(apiName))?.Scope;
```

Reply:
But it is just a suggestion - if you find it hard to read, we can leave it as it is

Reply:
I prefer if else structure :)